### DISCUSSION TITLE
PrismaClient is unable to run in this browser environment

### DISCUSSION THREAD

**Discussion by mayur4895** [prisma/prisma#24092] [Q&A]
### Bug description

Uncaught (in promise) Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `unknown`).

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by ludralph** [comment]
Hi @mayur4895 

Thank you for raising this question.

The error you're encountering is due to trying to run PrismaClient in a browser environment. PrismaClient is designed to run in a server-side environment and does not work on the client side as browsers do not execute SQL queries.
If you're using Next.js, there are a couple of solutions you can try:

- If you're using the Next.js app router, add "use server" at the top of your file.
- If you're using the Next.js pages router, keep the files that import Prisma or contain any Prisma query function inside the /api directory.

Remember, Prisma Client should be added to your Node.js app or inside the backend functions of Next.js.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
